lib/motsu-proc/tests/default.rs

```diff
@@ -0,0 +1,94 @@
+mod erc20 {
+    use alloy_primitives::U256;
+    use stylus_sdk::stylus_proc::sol_storage;
+    sol_storage! {
+        #[derive(motsu_proc::DefaultStorageLayout)]
+        pub struct Erc20 {
+            mapping(address => uint256) _balances;
+            mapping(address => mapping(address => uint256)) _allowances;
+            uint256 _total_supply;
+        }
+    }
+
+    #[motsu::test]
+    fn erc20_initiates_correctly(contract: Erc20) {
+        assert_eq!(contract._total_supply.get(), U256::from(0));
+    }
+}
+
+mod uint {
+    use alloy_primitives::{U128, U16, U256, U32, U64, U8};
+    use stylus_sdk::stylus_proc::sol_storage;
+
+    sol_storage! {
+        #[derive(motsu_proc::DefaultStorageLayout)]
+        pub struct UintContract {
+            uint256 a;
+            uint128 b;
+            uint64 c;
+            uint32 d;
+            uint16 e;
+            uint8 f;
+            uint8 g;
+        }
+    }
+
+    #[motsu::test]
+    fn uint_initiates_correctly(contract: UintContract) {
+        assert_eq!(contract.a.get(), U256::from(0));
+        assert_eq!(contract.b.get(), U128::from(0));
+        assert_eq!(contract.c.get(), U64::from(0));
+        assert_eq!(contract.d.get(), U32::from(0));
+        assert_eq!(contract.e.get(), U16::from(0));
+        assert_eq!(contract.f.get(), U8::from(0));
+        assert_eq!(contract.g.get(), U8::from(0));
+    }
+
+    #[motsu::test]
+    fn uint_updates_correctly(contract: UintContract) {
+        contract.a.set(U256::from(1));
+        contract.b.set(U128::from(2));
+        contract.c.set(U64::from(3));
+        contract.d.set(U32::from(4));
+        contract.e.set(U16::from(5));
+        contract.f.set(U8::from(6));
+        contract.g.set(U8::from(7));
+
+        assert_eq!(contract.a.get(), U256::from(1));
+        assert_eq!(contract.b.get(), U128::from(2));
+        assert_eq!(contract.c.get(), U64::from(3));
+        assert_eq!(contract.d.get(), U32::from(4));
+        assert_eq!(contract.e.get(), U16::from(5));
+        assert_eq!(contract.f.get(), U8::from(6));
+        assert_eq!(contract.g.get(), U8::from(7));
+    }
```

Comment:
Beautiful! This is what I had in mind!

I know it's a bit tedious, but could we also have a test case, where the types are unsorted, like

```
assert_eq!(contract.a.get(), U256::from(1));
assert_eq!(contract.b.get(), U16::from(5));
assert_eq!(contract.c.get(), U64::from(3));
assert_eq!(contract.d.get(), U8::from(6));
assert_eq!(contract.e.get(), U32::from(4));
assert_eq!(contract.f.get(), U128::from(2));
assert_eq!(contract.g.get(), U8::from(7));
```

Reply:
Sure, that makes sense! I tried to align the types to be more compact in storage(and thus stress testing more the algorithm), but I'll add a couple more for a contract where the types don't line up with storage the same way.